Dynatrace/dynatrace-operator
# Adds NO_PROXY feature-flag to the dynakube

# Description
Using the `feature.dynatrace.com/no-proxy` annotation on the `DynaKube` its possible to define the NO_PROXY of the dynatrace client.

**IMPORTANT:** This only effects the communications to the **dynatrace API** in the following components:
- operator pod
- csi-driver pod
- injected init-container (standalone init)

This PR also changes/generalises the way we get/read the proxy in the codebase (there was a lot of redundant code)
Also changed some names in an old tests. (dtclient tests)

## How can this be tested?
To test it properly is a bit cumbersome, so here is a tricky test:
 
- Get a cluster where you can restrict (via a NetworkPolicy) the dynatrace namespace, so its mandatory to use a proxy.
- Deploy a dynakube with a proxy. Make sure everything works as expected.
- Add your dyanatrace tenant to the NO_PROXY
- Now nothing should work :) (because the tenant is in the NO_PROXY, it wont be routed to the proxy, so the NetworkPolicy will not allow the request to go through -> reconcile fails)


## Checklist
